### PR TITLE
[14.0][IMP] barcodes_gs1: templates and expiry improvements

### PR DESCRIPTION
**Pull request by eantones**
Supersedes #326 #358

@FrankC013

### CODE REVIEW

**Comment by FrankC013** [comment]
@eantones Please review the changes

File: barcodes_gs1_label/report/report_barcode_gs1.py [hunk 1]
```diff
@@ -76,16 +76,18 @@ def _get_product_lot(self, products, quants, with_stock):
...
+        if lot:
+            if lot.product_id.tracking == "lot":
+                res["10"] = lot.name
+            elif lot.product_id.tracking == "serial":
+                if lot.ref:
+                    res["10"] = lot.ref
+                res["21"] = lot.name
         # elif product.tracking == "none":
         #     ???
         return res
```
Comment (eantones) [inline barcodes_gs1_label/report/report_barcode_gs1.py]:
```suggestion
    def _prepare_gs1_values(self, product, lot):
        if lot and lot.product_id != product:
            raise ValidationError(_"Incoherent....")
        res = {}
        if product.barcode:
            res["01"] = product.barcode.rjust(14, "0")
        if lot:
            if product.tracking == "lot":
                res["10"] = lot.name
            elif product.tracking == "serial":
                if lot.ref:
                    res["10"] = lot.ref
                res["21"] = lot.name  
```

File: barcodes_gs1_label/report/report_barcode_gs1.py [hunk 2]
```diff
@@ -248,7 +250,8 @@ def _get_report_values(self, docids, data=None):  # noqa: C901
                         res.append(fnc1)
                 doc["barcode_values"] = gs1_barcode
                 doc["barcode_string"] = requests.utils.quote("".join(res))
-                lot.gs1_generated = True
+                if lot:
+                    lot.gs1_generated = True
```
Comment (eantones) [inline barcodes_gs1_label/report/report_barcode_gs1.py]: Remove gs1_generated everywhere

**Review by eantones — COMMENTED** [review]
@FrankC013 changes requested

**Comment by FrankC013** [comment]
@eantones Please review the commits

**Pull request merged**